### ISSUE TITLE
Error parsing older angular UI with new 3.4.0 release

### ISSUE THREAD

**Issue by mjasplin** [twbs/bootstrap#27869]
We are using Angular 1.6, and we just ran into issues when upgrading from 3.3.7 to 3.4.0 bootstrap.  It seems to have issues with the #/linkName in the href field.  Reverted back to 3.3.7 and it is working fine. 

`<ul class="dropdown-menu">`
`<li><a href="#/amount" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Amount Type</a>`
`</li> </ul>`
  
Error appearing in console: 
bundle.js:16915 Uncaught Error: Syntax error, unrecognized expression: #
    at Function.Sizzle.error (bundle.js:16915)
    at Sizzle.tokenize (bundle.js:17572)
    at Sizzle.select (bundle.js:17993)
    at Function.Sizzle [as find] (bundle.js:16316)
    at jQuery.fn.init.find (bundle.js:18239)
    at getParent (bundle.js:101210)
    at HTMLAnchorElement.<anonymous> (bundle.js:101220)
    at Function.each (bundle.js:15812)
    at jQuery.fn.init.each (bundle.js:15584)
    at clearMenus (bundle.js:101218)

**Comment by rschwab** [comment]
This is happening on my development site as well, using Jquery 1.9 and/or 1.10.

Example:
https://jsbin.com/daqutaxace/edit?html,output

**Comment by Johann-S** [comment]
It fail because in your `data-target` you have a wrong selector (`#`).

In v3.3.7 to query our elements we did:
```js
$(selector)
```

But in v3.4.0 to avoid XSS, jQuery recommend to use
```js
$(document).find(selector)
```

and when `selector` is equal to `#` it throw an error.

A solution would be to change that line : https://github.com/twbs/bootstrap/blob/v3.4.0/js/dropdown.js#L32

To:

```js
var $parent = selector !== '#' && $(document).find(selector)

**Comment by XhmikosR** [comment]
@Johann-S should we close this now?

**Comment by Johann-S** [comment]
It depends if we want to release a fix or not here.

If not, for those who have this error you have to specify a selector in the `data-target` attribute to select your dropdown menu. See: https://jsbin.com/yumojudude/edit?html,output

**Comment by XhmikosR** [comment]
Not sure either. I mean, I know many people do still use 3.x.

**Comment by XhmikosR** [comment]
Should be fixed in the upcoming v3.4.1.
